### PR TITLE
Autofilter

### PR DESCRIPTION
**Pull request by alump**
I started autofiltering, there is one styling lint error, but until we get our theming done correctly I assume it's good enough for now. This will add autofilter field, and it has event handling. There is no actual filtering yet behind it, but hopefully good enough start.

I named this component as px-auto-filter-field as this might be separated from px-data-grid at some point.

I only did light theming variables for now, and anyway those are just some initial values.

### CODE REVIEW

**Comment by alump** [comment]
OK, cheated with lint testing... all theming needs to be anyway still adjusted later.

File: px-auto-filter-field.html [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  <template>
+    <style>
+
+      :host {
+        position: relative;
+        display: block;
+      }
+
+      .field {
+        background-color: var(--px-auto-filter-field-background-color, red);
```
Comment (limonte) [inline px-auto-filter-field.html]: you can exclude some stylelint rules here: https://github.com/vaadin/px-data-grid/blob/master/.stylelintrc

File: px-auto-filter-field.html [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+          input.addEventListener('keypress', event => this._onKeyPress(event));
+          input.addEventListener('change', event => this._onChange(event));
+        }
+
+        _onChange(event) {
+          this._updateValue(true);
+        }
+
+        _onKeyPress(event) {
+          if (event.keyCode === 13) {
```
Comment (limonte) [inline px-auto-filter-field.html]: `keyCode` is deprecated, use `event.key` instead. 

https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode

Also, for people from Japan,  `e.isComposing` is required.

So

```js
if (e.key === 'Enter' && !e.isComposing) {
```

File: px-auto-filter-field.html (hunk 1 above)
Comment (alump) [reply px-auto-filter-field.html]: To be fixed later.

**Pull request merged**